tools/azure-sdk-tools/packaging_tools/generate_client.py

```diff
@@ -40,19 +40,24 @@ def generate_autorest(folder: Path) -> None:
 
 
 def generate_typespec(folder: Path) -> None:
+    tsp_location_path = folder / "tsp-location.yaml"
 
-    # Turns out, "pwsh" is the name for PowerShell 7 on Windows, that is required for those scripts
-    ps_cmd = "pwsh"
+    if not tsp_location_path.exists():
+        raise ValueError(
+            "Didn't find a tsp_location.yaml in local directory. Please provide a valid "
+            "tsp-location.yaml file, for more information on how to create one, see: "
+            "https://github.com/Azure/azure-sdk-tools/tree/main/tools/tsp-client/README.md"
+        )
 
-    completed_process = run([ps_cmd, "../../../eng/common/scripts/TypeSpec-Project-Sync.ps1", folder], cwd=folder)
+    completed_process = run(["tsp-client", "update"], cwd=folder, shell=True, stderr=PIPE)
```

Comment:
Will take a look at why this was having issues without shell=True, but it is lower risk since we arent passing any parameters here